app/src/main/java/com/sanjeevyadavit/magecart/presentation/login/LoginViewModel.kt

```diff
@@ -0,0 +1,71 @@
+package com.sanjeevyadavit.magecart.presentation.login
+
+import android.app.Application
+import android.widget.Toast
+import androidx.lifecycle.*
+import com.sanjeevyadavit.magecart.common.IState
+import com.sanjeevyadavit.magecart.common.Resource
+import com.sanjeevyadavit.magecart.domain.use_case.LoginUseCase
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+@HiltViewModel
+class LoginViewModel @Inject constructor(
+    private val loginUseCase: LoginUseCase,
+    app: Application
```

Comment:
do not inject application here. You can inherit ViewModel class and you wont need this